apollo/command_to_gcode.py

```diff
@@ -0,0 +1,145 @@
+import asyncio
+
+from apollo.smoothie import SmoothieCom
+from config.settings import Config
+
+
+class CommandToGCode(object): 
+    """
+    this module converts front-end commands to GCode
+    then sends GCode strings to Smoothie-Com module
+    """
+
+    def __init__(self):
+        self.smoothie_com = SmoothieCom(Config.SMOOTHIE_URL, Config.SMOOTHIE_PORT)
+
+    @asyncio.coroutine
+    def connect(self):
+        """
+        tell the smoothie module to connect to the motor controller
+        return True if successful
+        """
+        return (yield from self.smoothie_com.connect())
+
+    @asyncio.coroutine
+    def send_gcode(self,string):
+        """
+        send a GCode string to the smoothie module
+        """
+        if len(string):
+            return (yield from self.smoothie_com.send(string))
+
+    @asyncio.coroutine
+    def process(self, command):
+        """
+        receive front-end commands, and convert them to GCode
+        returns None or a coordinate dict
+        """
+
+        # check to see if this module has a method matching the command's 'type'
+        method = getattr(self, command.get('type', '') , None)
+
+        if method:
+
+            coords = None
+
+            # pass the data to the matching method, returning an array of gcode strings
+            for line in method( command.get('data',{}) ):
```

Comment:
too many things going on in this line..

```
command_args = command.get('data', {})

gcodes = gcode_handler(command_args)

for gcode in gcodes:
    .....
```

Also the variable `line` isn't too meaningful....